mongo/src/main/java/com/redhat/lightblue/mongo/metadata/BSONParser.java

```diff
@@ -132,14 +126,52 @@ public Object asRepresentation(Object value) {
     }
 
     /**
-     * Override to set _id appropriately.
+     * Override to convert partialFilterExpression index property from string to map.
+     *
+     */
+    @Override
+    public EntityInfo parseEntityInfo(Object object) {
```

Comment:
You can simplify this by overriding parseIndex. Then it'll be called for each index.